docs/source/workflows/design_spaces.rst

```diff
@@ -158,3 +155,84 @@ The example code below creates a registers a design space based on this Gem Tabl
     )
 
     registered_design_space = project.design_spaces.register(design_space)
+
+Formulation Design Space
+------------------------
+
+A formulation design space defines the set of formulations that can be produced from a given set of ingredient names, labels and constraints.
+Ingredient names are specified as a set of strings, where each string is a unique ingredient name, e.g., ``{'water', 'salt'}`` to a design space with two ingredients.
+Labels are specified as a mapping from each label to a set of ingredient names that should be given that label is present in a formulation, e.g., ``{'solute': {'salt'}}``.
+An ingredient may be given multiple labels, and these labels are static.
+An ingredient will always be given all applicable labels when present in a formulation.
```

Comment:
How does this work with multi-level formulations? If ingredient "binder3" has label "binder" but it is only included in the formulation indirectly through "binder solution4", then when the formulation is flattened to include "binder3" will it be labeled?

Reply:
In your example, the label for "binder3" will be added by a `SimpleMixturePredictor` when the formulation is flattened. The design space doesn't know that "binder3" is present in "binder solution4", so there won't be a "binder" label in the un-flattened formulation.

Reply:
ok, that is how I would expect it to work